src/crimson/os/seastore/transaction.h

```diff
@@ -284,12 +285,17 @@ class Transaction {
     assert(extent.get_paddr() == placeholder.get_paddr());
     assert(extent.get_paddr().is_absolute());
     {
-      auto where = read_set.find(placeholder.get_paddr());
+      auto where = read_set.find(placeholder.get_paddr(), extent_cmp_t{});
       assert(where != read_set.end());
       assert(where->ref.get() == &placeholder);
       where = read_set.erase(where);
-      auto it = read_set.emplace_hint(where, this, &extent);
+      read_items.emplace_back(this, &extent);
+      auto it = read_set.insert_before(where, read_items.back());
       extent.read_transactions.insert(const_cast<read_set_item_t<Transaction>&>(*it));
+#ifndef NDEBUG
+      num_replace_placeholder++;
+      assert(read_set.size() + num_replace_placeholder == read_items.size());
```

Comment:
See https://github.com/ceph/ceph/pull/62878#discussion_r2074978222

I don't think this is the correct place to assume all the wait-io() must be finished, so that the assert holds.

The only place I can think of that all the read wait-io() must be finished is the `prepare_record()` time.